sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -52,7 +52,8 @@ class UnivocityParser(
     requiredSchema: StructType,
     val options: CSVOptions,
     filters: Seq[Filter]) extends Logging {
-  require(requiredSchema.toSet.subsetOf(dataSchema.toSet),
+  require(requiredSchema.toSet.subsetOf(
+    CharVarcharUtils.removeCharVarcharTypeStringMetadataKey(dataSchema).toSet),
```

Comment:
instead of removing the metadata key, shall we restore the actual schema (with char/varchar type present) before checking?

Reply:
Let me investigate it first.